### PR TITLE
mon: PG Monitor should report waiting for backfill

### PR DESCRIPTION
**Pull request by theanalyst**
Instead of backfill (as the state is `wait_backfill`)

Fixes: #12744

### CODE REVIEW

File: src/mon/PGMonitor.cc [hunk 1]
```diff
@@ -2061,7 +2061,7 @@ void PGMonitor::get_health(list<pair<health_status_t,string> >& summary,
     if (p->first & PG_STATE_INCOMPLETE)
       note["incomplete"] += p->second;
     if (p->first & PG_STATE_BACKFILL_WAIT)
-      note["backfill"] += p->second;
+      note["waiting for backfill"] += p->second;
```
Comment (tchaikov) [inline src/mon/PGMonitor.cc]: @theanalyst 

should be "backfill-wait" instead? to be consistent with the document.

or, we should be consistent with the naming of "foo-wait", like "recovery_wait". so we should update both the doc and the code:

- s/backfill-wait/backfill_wait/ in doc
- s/"backfill"/"backfill_wait"/ in the line above

Comment (theanalyst) [inline src/mon/PGMonitor.cc]: I guess I'll change to backfill_wait here and the doc as well..

**Comment by theanalyst** [comment]
@tchaikov updated

**Comment by tchaikov** [comment]
lgtm modulo the typo in commit message.

**Comment by theanalyst** [comment]
Updated again. Thanks for the patient reviews

**Comment by tchaikov** [comment]
lgtm

**Pull request merged**